endpoint.cpp

```diff
@@ -545,6 +545,7 @@ TcpEndpoint::TcpEndpoint()
 
 TcpEndpoint::~TcpEndpoint()
 {
+    close();
```

Comment:
ok... now checking again, `fd` is from the parent class. So... yes. We could just log the message on the destructor.  However I think it makes sense to do inside the close() method since it would catch all cases in which the connection is actually closed.  Thoughts?